### PR TITLE
fix: move karpenter ECR auth token and provider out of the module

### PR DESCRIPTION
**Pull request by oleksiimorozenko**
# [Pull request title](fix: move karpenter ECR auth token and provider out of the module)

## Description

Defining provider and related resources (data source) in the module can break even plan if the user running Terraform is assuming role (i.e. doesn't have any permissions assigned to itself but assume role) even if Karpenter module is disabled

The plan fails with `Error: getting ECR Public authorization token: AccessDeniedException: User: arn:aws:iam::<account-id>:user/username is not authorized to perform: ecr-public:GetAuthorizationToken on resource: * because no identity-based policy allows the ecr-public:GetAuthorizationToken action` in this case

It's better instead to define the provider in the root module as the [dependent module's example](https://github.com/terraform-aws-modules/terraform-aws-eks/blob/6c56e2ad20057a5672526b5484df96806598a4e2/examples/karpenter/main.tf#L51) shows
This PR solves the problem when the provider block looks like this:
```terraform
provider "aws" {
  region = var.aws_region
  assume_role {
    session_name = "terraform-role"
    role_arn     = "arn:aws:iam::<account-id>:role/rolename"
  }
}
```

### Checklist

- [ ] CI tests are passing
- [ ] README.md has been updated after any changes to variables and outputs. See https://github.com/particuleio/terraform-kubernetes-addons/#doc-generation

### CODE REVIEW

**Comment by oleksiimorozenko** [comment]
@ArchiFleKs @rguichard any chance it will be approved?

**Comment by rguichard** [comment]
> @ArchiFleKs @rguichard any chance it will be approved?

Don't worry we will review it! Thx a lot for your contribution!

**Review by ArchiFleKs — APPROVED** [review]

**Review by ArchiFleKs — CHANGES_REQUESTED** [review]

File: modules/aws/.terraform-docs.yml [hunk 1]
```diff
@@ -1,2 +0,0 @@
-settings:
-  lockfile: false
```
[the remark this answers is not in the document]
Comment (oleksiimorozenko) [reply modules/aws/.terraform-docs.yml]: Seems like I've deleted it accidentally
Put it back already

**Review by ArchiFleKs — APPROVED** [review]

**Comment by ArchiFleKs** [comment]
@oleksiimorozenko could you rebase please

**Pull request merged**